### PR TITLE
fix(ci): upgrade create-github-app-token to v3 and fix permission inputs

### LINKED ISSUE #1960: CI: gitleaks version updater fails — create-github-app-token@v1 rejects client-id
## Summary

The scheduled `gitleaks version updater` workflow fails at the token-minting step:

```
Error: Input required and not supplied: app-id
```

Failing run: https://github.com/lightspeedwp/.github/actions/runs/31991525772

## Root cause

`.github/workflows/gitleaks-update.yml` passes `client-id`, but the action is pinned to `@v1`, which only accepts `app-id`. The runner emits `Unexpected input(s) 'client-id'` and the action aborts. The same run raises the Node 20 deprecation annotation, since v1 targets `node20`.

## Related breakage found while investigating

- `permissions-contents` / `permissions-issues` are misspelled in two workflows (correct prefix is `permission-`), so those minted tokens are not least-privilege scoped.
- Three gitlink entries under `agents/chat-closure-agent/tests/fixtures/integration-e2e/` have no `.gitmodules` entries, so `actions/checkout` aborts on every PR.
- Four workflows hard-code `node-version` rather than reading `.nvmrc`; branch name validation fails on Node 20 because markdownlint requires >= 22.

### PR DESCRIPTION
**Pull request by eleshar**
# Bugfix Pull Request

## Linked issues

Fixes #1960

## Context

- Severity/Impact: High — the nightly gitleaks version updater has been failing on every scheduled run, and a separate submodule fault fails checkout on every pull request.
- Affected versions/environments: `lightspeedwp/.github` on `main`, GitHub-hosted `ubuntu-latest` runners.

## Reproduction

- Steps: 1) Wait for (or dispatch) the `gitleaks version updater` schedule. 2) Observe the `Mint a short-lived App token` step. 3) Job aborts before checkout.
- Expected vs Actual: Expected a short-lived App token and an update PR; actual is `Error: Input required and not supplied: app-id`.

Failing run: https://github.com/lightspeedwp/.github/actions/runs/31991525772

## Root Cause

1. `gitleaks-update.yml` passes `client-id` to `actions/create-github-app-token@v1`. That input only exists from v2 onwards, so the runner warned `Unexpected input(s) 'client-id'` and the action exited on its own `app-id` required check. v1 also targets `node20`, producing the Node 20 deprecation annotation on the same run.
2. `permissions-contents` / `permissions-issues` are misspelled (the action's inputs are `permission-*`), so the minted tokens were never scoped down and silently carried the App installation's full permissions.
3. `issue-fields-backfill.yml` fell back to the numeric `LS_APP_ID` in the `client-id` slot. An App ID is not a Client ID, so that fallback could never mint a token.
4. Three gitlinks under `agents/chat-closure-agent/tests/fixtures/integration-e2e/` were committed without `.gitmodules` entries (leftovers from a local test run in #1933), so `actions/checkout` aborts with `fatal: No url found for submodule path …` on every PR.
5. Four workflows hard-code `node-version` instead of reading `.nvmrc`. Branch name validation failed on Node 20 with `npm error EBADENGINE … markdownlint@0.41.1 Required: {"node":">=22"}`.

## Fix Summary

- Upgraded all four `create-github-app-token` callers to v3, which accepts `client-id` and runs on `node24`. `gitleaks-update.yml` is SHA-pinned (`bcd2ba4` = v3.2.0) to match that file's existing convention; the others use tag refs, matching theirs.
- Corrected `permissions-*` → `permission-*` in `issue-fields-backfill.yml` and `metadata-governance.yml`, restoring the intended least-privilege scoping.
- `issue-fields-backfill.yml` now passes `client-id` when `LS_APP_CLIENT_ID` is set and falls back to `app-id` otherwise.
- Removed the three stray gitlinks; `workspace-cleaner.test.js` creates its own temporary repositories, so nothing referenced them.
- Pointed the four hard-coded workflows at `node-version-file: ".nvmrc"` (Node 22, the repository pin).

## Verification

- [ ] Tests added/updated to cover the bug — n/a, workflow configuration only
- [x] YAML parses for all eight modified workflows
- [x] `v3.2.0` `action.yml` confirmed upstream: `client-id` present, `runs.using: node24`, permission inputs named `permission-*`
- [x] Confirmed `.nvmrc` is `22`, satisfying markdownlint's `node >= 22` engine constraint
- [x] Confirmed the three gitlinks are unreferenced and were introduced by the immediately preceding merge (#1933)
- [ ] End-to-end verification of `gitleaks-update.yml` requires the org App credentials and therefore happens on the next scheduled or dispatched run

## Risk & Rollback

- Risk level: Low — configuration-only, no runtime code paths touched.
- Rollback plan: revert this PR.

### Fixed

- `actions/create-github-app-token` upgraded to v3 across all four callers, restoring the gitleaks version updater and clearing the Node 20 deprecation annotation. (Fixes #1960)
- Misspelled `permissions-*` token inputs corrected to `permission-*`, restoring least-privilege scoping on minted App tokens. (Fixes #1960)
- Stray submodule pointers removed from `agents/chat-closure-agent/tests/fixtures/integration-e2e/`, restoring `actions/checkout` on pull requests. (Fixes #1960)
- Four workflows now read `.nvmrc` instead of hard-coding `node-version`. (Fixes #1960)

---

### Checklist (Global DoD / PR)

- [x] All AC met and demonstrated
- [x] Tests added/updated (unit/E2E as appropriate) — n/a, workflow configuration only; there is no test harness for GitHub Actions YAML in this repo
- [x] Accessibility checklist completed (where relevant) — n/a, no user-facing surface
- [x] Docs/readme/changelog updated (if user-facing)
- [x] Security checklist completed (where relevant):
  - [x] Untrusted input validated and sanitised — no new untrusted input reaches a shell
  - [x] Output escaped for its rendering context — n/a
  - [x] Privileged actions enforce nonce and capability checks — App token permissions now correctly scoped down
  - [x] No secrets/sensitive data introduced; OWASP risks reviewed
- [x] Code/design reviews approved — pending reviewer sign-off on this PR
- [x] CI green; linked issues closed; release notes prepared (if shipping) — satisfied at merge; #1960 closes automatically

---

### CODE REVIEW

File: .github/workflows/issue-fields-backfill.yml [hunk 1]
```diff
@@ -75,13 +75,16 @@ jobs:
...
+          # LS_APP_CLIENT_ID is preferred; LS_APP_ID is the legacy numeric App ID
+          # and must be passed as app-id, not as a client-id fallback.
+          client-id: ${{ vars.LS_APP_CLIENT_ID }}
+          app-id: ${{ vars.LS_APP_CLIENT_ID == '' && vars.LS_APP_ID || '' }}
           private-key: ${{ secrets.LS_APP_PRIVATE_KEY }}
           repositories: ${{ github.repository }}
-          permissions-contents: write
-          permissions-issues: write
+          permission-contents: write
+          permission-issues: write
```
[the remark this answers is not in the document]
Comment (eleshar) [reply .github/workflows/issue-fields-backfill.yml:92]: Applied: dropped `permission-contents` (the checkout uses the default `GITHUB_TOKEN`) and added `permission-organization-projects: write`, which the Projects v2 mutations in this job require.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: .github/workflows/project-meta-sync.yml [hunk 2]
```diff
@@ -66,7 +66,7 @@ jobs:
       - name: Create GitHub App installation token
         id: app-token
         if: steps.preflight.outputs.enabled == 'true'
-        uses: actions/create-github-app-token@v1
+        uses: actions/create-github-app-token@v3
         with:
           app-id: ${{ vars.LS_APP_ID }}
           private-key: ${{ secrets.LS_APP_PRIVATE_KEY }}
```
[the remark this answers is not in the document]
Comment (eleshar) [reply .github/workflows/project-meta-sync.yml:72]: Applied: `project-meta-sync` now passes explicit `permission-organization-projects`, `permission-issues` and `permission-pull-requests` instead of inheriting every permission the App installation holds.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: CHANGELOG.md [hunk 3]
```diff
@@ -53,6 +53,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
...**`actions/create-github-app-token` upgraded to v3 — gitleaks version updater restored** — The nightly gitleaks version updater failed with `Error: Input required and not supplied: app-id` because the workflow passed `client-id`, an input that only exists from v2 onwards, to the v1 action. Upgraded all four callers (`gitleaks-update.yml`, `issue-fields-backfill.yml`, `metadata-governance.yml`, `project-meta-sync.yml`) to v3, which supports `client-id` and runs on Node 24 (current LTS), clearing the Node 20 deprecation annotation. `gitleaks-update.yml` is SHA-pinned (`bcd2ba4`, v3.2.0) to match that workflow's pinning convention. Also corrected the misspelled `permissions-contents` / `permissions-issues` inputs to `permission-contents` / `permission-issues` in `issue-fields-backfill.yml` and `metadata-governance.yml`, which had been silently ignored, leaving the minted tokens without the intended least-privilege scoping. Finally, `issue-fields-backfill.yml` no longer passes the legacy numeric App ID as a `client-id` fallback — the two are different identifiers, so that path could never mint a token; it now passes `client-id` when `LS_APP_CLIENT_ID` is set and falls back to `app-id`.
```
[the remark this answers is not in the document]
Comment (eleshar) [reply CHANGELOG.md]: Applied: all six new `[Unreleased]` → `Fixed` entries now carry `([PR #1959](https://github.com/lightspeedwp/.github/pull/1959), [#1960](https://github.com/lightspeedwp/.github/issues/1960))`.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: .github/workflows/checks.yml [hunk 4]
```diff
@@ -29,16 +29,37 @@ jobs:
...
+    env:
+      # lint:md:changed diffs this range; without it the base ref cannot be
+      # resolved from a shallow checkout and the markdown lint would silently
+      # skip. fetch-depth: 0 supplies the history the diff needs.
+      BASE_SHA: ${{ github.event.merge_group.base_sha || github.event.pull_request.base.sha || github.event.before }}
+      HEAD_SHA: ${{ github.event.merge_group.head_sha || github.event.pull_request.head.sha || github.sha }}
     steps:
       - uses: actions/checkout@v7
+        with:
+          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (eleshar) [reply .github/workflows/checks.yml:41]: Applied: added `persist-credentials: false` to all three `actions/checkout` steps in `checks.yml` (Lint, Test, Validate) — none of the three jobs push, commit, or need authenticated git access after checkout.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: scripts/validation/lib/changed-files.cjs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  } catch {
+    // Also covers the all-zero SHA GitHub sends as github.event.before on the
+    // first push to a branch, which resolves to nothing.
+    return null;
+  }
+}
+
+function resolveRange() {
+  const baseSha = commitSha(process.env.BASE_SHA);
+  const headSha = commitSha(process.env.HEAD_SHA);
```
[the remark this answers is not in the document]
Comment (eleshar) [reply scripts/validation/lib/changed-files.cjs:30]: Verified and dismissed: `npx eslint scripts/validation/lib/changed-files.cjs scripts/validation/lint-md-changed.cjs scripts/validation/validate-workflow-npm-scripts.cjs` against this repo's actual `eslint.config.cjs` (the `**/*.cjs` block) reports zero errors for all three files — matching the real Linting job, which has already passed multiple times with these exact files present. This appears to be CodeRabbit's own supplementary lint pass not resolving the project's flat config, rather than a real issue.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: CHANGELOG.md [hunk 6]
```diff
@@ -36,7 +36,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
...neration validation (JSON/CSV/Markdown), stale issue detection accuracy, and data integrity checks (orphaned/conflicting/duplicate labels). Implemented `staging-validation.js` script (400+ lines) providing modular validation tasks with CLI interface supporting `--all` (run all tests), `--task <name>` (individual task execution), `--count <n>` (configurable issue count), and structured JSON reporting with GO/NO-GO production readiness decision. Created test data fixtures (`staging-test-data.json`) with 100 representative test issues covering 7 categories: issue types (10 each type), age distribution (recent/active/aging/stale), PR relationships, label scenarios (correct/missing/conflicting), comment density, and edge cases (unicode, emoji, long content, special chars, locked/archived issues). Success criteria defined: 95%+ audit accuracy, < 5 minute execution for 100 issues, < 0.5% error rate, 100% data consistency, zero critical errors. All test infrastructure ready for manual validation runs against staging environment. ([Phase 5.2 Project](.github/projects/active/issue-maintenance-phase-5-2-staging-2026-08-12/), [Depends: #1780](https://github.com/lightspeedwp/.github/pull/1780))
```
[the remark this answers is not in the document]
Comment (eleshar) [reply CHANGELOG.md:39]: Skipped: this entry is pre-existing historical content (Phase 5.2 Staging Validation) that I only touched to fix its relative project-path prefix in an earlier commit — I didn't author the entry or its links. GitHub's API is currently degraded (503s observed independently from GraphQL and REST in this same session), so I can't reliably look up the correct tracking issue number to add right now, and guessing one would be worse than leaving the gap. Happy to add it in a follow-up once the correct issue number can be confirmed.
